src/migratus/core.clj

```diff
@@ -68,6 +68,25 @@
   (let [completed? (set (proto/completed-ids store))]
     (filter (comp completed? proto/id) (mig/list-migrations config))))
 
+(defn gather-migrations 
+  "Returns a list of all migrations from migration dir and db
+  with enriched data:
+    - date and time when was applied;
+    - description;"
+  [config store]
+  (let [completed-migrations (vec (proto/completed store))
+        available-migrations (mig/list-migrations config)
+        merged-migrations-data (apply merge completed-migrations available-migrations)
+        grouped-migrations-by-id (group-by :id merged-migrations-data)]
+    (map (fn [[_ v]] (apply merge v)) grouped-migrations-by-id)))
```

Comment:
it would make the code clearer if you give the fn a name .

Reply:
named fn